### ISSUE TITLE
Supervised Loss

### ISSUE THREAD

**Issue by khs6848** [jsyoon0823/TimeGAN#15]
Hi, 

Would you be able to elaborate the following as to why H[:,1:,:], H_hat_supervise[:,:-1,:]?
G_loss_S = tf.losses.mean_squared_error(H[:,1:,:], H_hat_supervise[:,:-1,:])

I think if we were to deal with uni-directional time-series (such as a financial data), shouldn't it be the following so that we don't look ahead?
G_loss_S = tf.losses.mean_squared_error(H[:,:-1, :], H_hat_supervise[:,1:,:])

Thanks and hope to hear back from you soon!

**Comment by jsyoon0823** [comment]
Thank you for your interests in our paper.

The main objective of the supervisor is "Generate next sequence using the previous sequence".
Therefore, G_loss_S forces the supervisor to generate next sequence (H[:,1:,:]) using previous sequence (H_hat_supervise[:,:-1,:]).
Therefore, the generated samples (H_hat_supervise) should be matched with future H.

Thanks

**Comment by zzw-zwzhang** [comment]
# 2. Supervised loss
G_loss_S = tf1.losses.mean_squared_error(H[:,1:,:], H_hat_supervise[:,:-1,:])

# Train generator       
_, step_g_loss_s = sess.run([GS_solver, G_loss_S], feed_dict={Z: Z_mb, X: X_mb, T: T_mb})

Why  G_loss_S can be used to train generator?
